src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -277,6 +277,20 @@ impl ScalarFunc {
         }
         Ok(Some(result))
     }
+
+    #[inline]
+    pub fn date_diff(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let l_t: Cow<Time> = try_opt!(self.children[0].eval_time(ctx, row));
+        if l_t.is_zero() {
+            return handle_incorrect_datetime_error(ctx, l_t).map(|_| None);
+        }
+        let r_t: Cow<Time> = try_opt!(self.children[1].eval_time(ctx, row));
+        if r_t.is_zero() {
+            return handle_incorrect_datetime_error(ctx, r_t).map(|_| None);
+        }
+        let days_diff = l_t.get_time().signed_duration_since(r_t.get_time()).num_days();
```

Comment:
What's the result of `("2018-02-02 00:00:00", "2018-02-01 12:00:00")`?

Reply:
Two follow-up questions:
1) In the Go implementation, it's start_date - end_date is it correct?
https://github.com/pingcap/tidb/blob/32b1dbd8d59fbd725ec3f2971da2323cae0bf885/types/mytime.go#L195
2) https://www.w3schools.com/sql/func_sqlserver_datediff.asp
Based on SQL specification it's end_date - start_date and user can select interval(e.g. year, month, day... ) Are we going to support interval?

Reply:
From the [description from MySQL](https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_datediff): "Only the date parts of the values are used in the calculation. "

Here is the output of MySQL:

```
mysql> SELECT DATEDIFF('2018-02-02 00:00:00', '2018-02-01 12:00:00');
+--------------------------------------------------------+
| DATEDIFF('2018-02-02 00:00:00', '2018-02-01 12:00:00') |
+--------------------------------------------------------+
|                                                      1 |
+--------------------------------------------------------+
1 row in set (0.00 sec)
```

So I think there is a misunderstanding here?

Reply:
"DATEDIFF() returns expr1 − expr2"  makes sense to me, the naming in KiDB is confusing. Thanks!